Add a note about docker installed from snap (#19309)
github/docs: content/actions/using-github-hosted-runners/about-github-hosted-runners.md

- `/github/workspace` - {% data reusables.repositories.action-root-user-required %}
- `/github/workflow`

### Docker engine setup

> To pass the inputs of an action into the docker container the runner uses the environment variables which may contain dashes as part of their names. If  the docker engine is not a binary executable but a shell wrapper or a link (a docker engine istalled on Linux with snap is a sample) the action is not able to get the inputs and the build fails with the message `Error: Input required and not supplied: java-version`.

Reviewer: ```suggestion
To pass the inputs of an action into the docker container, the runner uses the environment variables, which may contain dashes as part of their names. The action may not able to get the inputs if the Docker engine is not a binary executable, and is instead a shell wrapper or a link (for example, a Docker engine installed on Linux using `snap). If this occurs, the build will subsequently fail with the message `Error: Input required and not supplied: java-version`.
```

> To pass the inputs of an action into the docker container the runner uses the environment variables which may contain dashes as part of their names. If  the docker engine is not a binary executable but a shell wrapper or a link (a docker engine istalled on Linux with snap is a sample) the action is not able to get the inputs and the build fails with the message `Error: Input required and not supplied: java-version`.

Reviewer: ```suggestion
To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. If the Docker engine is not a binary executable but a shell wrapper or a link (for example, a Docker engine installed on Linux with `snap`) the action is not able to get the inputs, and the build consequently fails with the message `Error: Input required and not supplied: java-version`.
```

- `/github/workspace` - {% data reusables.repositories.action-root-user-required %}
- `/github/workflow`

### Docker engine setup

> To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. If the Docker engine is not a binary executable but a shell wrapper or a link (for example, a Docker engine installed on Linux with `snap`) the action is not able to get the inputs, and the build consequently fails with the message `Error: Input required and not supplied: java-version`.

Reviewer: ```suggestion
To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. The action may not able to get the inputs if the Docker engine is not a binary executable, but is instead a shell wrapper or a link (for example, a Docker engine installed on Linux using `snap). If this occurs, the build will subsequently fail with the message `Error: Input required and not supplied: java-version`.
```

> ### Docker engine setup
> 
> To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. The action may not able to get the inputs if the Docker engine is not a binary executable, but is instead a shell wrapper or a link (for example, a Docker engine installed on Linux using `snap). If this occurs, the build will subsequently fail with the message `Error: Input required and not supplied: java-version`.

Reviewer: ```suggestion
### Checking how Docker was installed

To pass the inputs of an action into the Docker container, the runner uses environment variables that might contain dashes as part of their names. The action may not able to get the inputs if the Docker engine is not a binary executable, but is instead a shell wrapper or a link (for example, a Docker engine installed on Linux using `snap`). If this occurs, the build will subsequently fail with the message `Error: Input required and not supplied: java-version`.
```
